website/content/en/docs/setup/installation/operating-systems/nixos.md

```diff
@@ -8,22 +8,50 @@ weight: 6
 [NixOS] is a Linux distribution built on top of the Nix package manager. This
 page covers installing and managing Vector on NixOS.
 
-Nixpkgs has a [community maintained package][nixpkg-vector] for Vector. It can
-be installed on a NixOS system with the following snippet in
-`configuration.nix`:
+Nixpkgs has a [community maintained module][nixpkg-vector] for Vector.
+The options for which may be viewed on the [NixOS Search][nixos-search].
+
+This can be used to deploy and configure Vector on a NixOS system.
+For example, place into a system's `configuration.nix`:
 
 ```nix
-environment.systemPackages = [
-  pkgs.vector
-];
+services.vector = {
+  enable = true;
+  journaldAccess = true;
+  settings = builtins.fromTOML ''
+    [sources.journald]
+    type = "journald"
+    current_boot_only = true
+
+    [sources.vector_metrics]
+    type = "internal_metrics"
+
+    [sinks.loki]
+    type = "loki"
+    inputs = [ "journald" ]
+    endpoint = "https://loki.mycompany.com"
+
+    [sinks.loki.labels]
+    source = "journald"
+
+    [sinks.prometheus_exporter]
+    type = "prometheus_exporter"
+    inputs = [ "vector_metrics" ]
+    address = "[::]:9598"
+  '';
+};
 ```
 
+The module will allow validate that the Vector configuration is valid before
```

Comment:
```suggestion
The module will also verify that the Vector configuration is valid before
```
To avoid using "validate" twice, and fix incorrect(?) word